qt/aqt/main.py

```diff
@@ -893,17 +892,23 @@ def setupAddons(self, args: list | None) -> None:
 
         if not self.safeMode:
             self.addonManager.loadAddons()
-            self.maybe_check_for_addon_updates()
 
-    def maybe_check_for_addon_updates(self) -> None:
+    def maybe_check_for_addon_updates(
+        self, on_done: Callable[[list[DownloadLogEntry]], None] | None = None
+    ) -> None:
         last_check = self.pm.last_addon_update_check()
         elap = int_time() - last_check
 
+        def wrap_on_updates_installed(log: list[DownloadLogEntry]) -> None:
+            if on_done:
+                on_done(log)
+            self.on_updates_installed(log)
+
         if elap > 86_400 or self.pm.last_run_version() != point_version():
             check_and_prompt_for_updates(
                 self,
                 self.addonManager,
-                self.on_updates_installed,
+                wrap_on_updates_installed,
```

Comment:
This skips calling on_done when time hasn't elapsed. Not a big deal in the grand scheme of things, but it's inconsistent with the auto sync code that always calls the callable.